tools/clang/lib/Sema/SemaExpr.cpp

```diff
@@ -5288,7 +5288,10 @@ Sema::BuildResolvedCallExpr(Expr *Fn, NamedDecl *NDecl,
   if (FDecl) {
     if (CheckFunctionCall(FDecl, TheCall, Proto))
       return ExprError();
-
+    if (CheckHLSLFunctionCall(FDecl, TheCall))
+      return ExprError();
+    if (hlsl::IsIntrinsicOp(FDecl) && CheckHLSLIntrinsicCall(FDecl, TheCall))
+      return ExprError();
```

Comment:
What do you mean? Is it called below it, only gated by the `IsInstrinsicOp` call.
Am I missing something?

Reply:
I meant that we could have one hook function into SemaHLSL (CheckHLSLFunctionCall), and the knowledge of using IsIntrinsicOp and CheckHLSLIntrinsicCall would be localized to CheckHLSLFunctionCall in SemaHLSL, instead of requiring this to be known in SemaExpr.  Also, CheckHLSLIntrinsicCall wouldn't have to be part of Sema.  That is unless you can think of a place that CheckHLSLIntrinsicCall would need to be called from Sema without calling CheckHLSLFunctionCall.

Reply:
Ah right, now I understand the 'under' 😊 Moved the call inside the CheckHLSLFunctionCall!